Shanghai Blood Sports/Assets/Scripts/Events/EventHandler.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Assets.Scripts.Events {
+    public class EventHandler {
+        public void onEvent<EVENT, ARGS>(EVENT e)
+            where EVENT : Event<ARGS>
+            where ARGS : EventArgs {
+            if (e is HitEvent hit) {

```

Comment:
https://stackoverflow.com/questions/298976/is-there-a-better-alternative-than-this-to-switch-on-type/299001#299001

Still not sold on if/switch case but this is probably more elegant